File: 프로그래머스/lv2/42885. 구명보트/구명보트.py
def solution(people, limit):
    answer = 0
    people.sort(reverse = True)
    # 1명 또는 2명씩 태우기
    a = 0
    b = len(people)-1
    while a <= b:
        if people[a] + people[b] <= limit:
            b -= 1
        a += 1
        answer += 1
    
    return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> people, int limit) {
    sort(people.begin(), people.end());
    int answer = 0;
    int p_left = 0;
    int p_right = people.size() - 1;
    while (p_left <= p_right) {
        int weight1 = people[p_left];
        int weight2 = people[p_right];
        if (weight1 + weight2 <= limit) {
            p_left++;
            p_right--;
        } else {
            p_right--;
        }
        answer++;
    }
    
    return answer;
}